component-api/src/main/java/org/talend/sdk/component/api/service/connection/CreateConnection.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (C) 2006-2021 Talend Inc. - www.talend.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.api.service.connection;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.talend.sdk.component.api.meta.Documentation;
+import org.talend.sdk.component.api.service.ActionType;
+
+@ActionType(value = "createconnection", expectedReturnedType = Object.class)
+@Target(METHOD)
+@Retention(RUNTIME)
+@Documentation("This class marks an action creating a connection")
+public @interface CreateConnection {
```

Comment:
Extends documentation and specify that this is _studio only_
If this annotation is intended to be implemented in a service class, please add a some validation if in/out classes inherit RuntimeContextInjector.